include/revng/Support/GraphAlgorithms.h

```diff
@@ -196,3 +201,215 @@ auto exitless_scc_range(NodeTy Entry) {
 
   return make_filter_range(Range, Filter);
 }
+
+namespace detail {
+
+template<typename T>
+T stripInverseHelper(llvm::Inverse<T> &&);
+
+template<typename T>
+T stripInverseHelper(T &&);
+
+template<typename T>
+using StripInverse = decltype(stripInverseHelper(std::declval<T>()));
+
+// clang-format off
+template<typename T>
+concept SupportedGraph = std::is_pointer_v<T> ||
+                         (StrictSpecializationOf<T, llvm::Inverse> &&
+                          std::is_pointer_v<StripInverse<T>>);
+// clang-format on
+
+template<typename GraphType>
+using GT = llvm::GraphTraits<GraphType>;
+
+} // namespace detail
+
+template<detail::SupportedGraph RawGraphType>
+class GraphNodeOrRootHelper {
```

Comment:
`GraphNodeOrVirtualRoot`?

Reply:
Closing, this since I found a way to get rid of this without losing functionality.